File: java/src/main/java/com/horizontal/java/adapter/HorizontalRecyclerAdapter.java
package com.horizontal.java.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.RequestManager;
import com.bumptech.glide.request.RequestOptions;
import com.horizontal.java.R;

import java.util.List;

public class HorizontalRecyclerAdapter extends RecyclerView.Adapter<HorizontalRecyclerAdapter.HorizontalRecyclerItemViewHolder> {

    private List<String> data;
    private RequestManager requestManager;

    public HorizontalRecyclerAdapter(List<String> data,RequestManager requestManager) {
        this.data = data;
        this.requestManager = requestManager;
    }

    @NonNull
    @Override
    public HorizontalRecyclerItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new HorizontalRecyclerItemViewHolder(View.inflate(parent.getContext(),R.layout.item_recycler_view,null));
    }

    @Override
    public void onBindViewHolder(@NonNull HorizontalRecyclerItemViewHolder holder, int position) {
        RequestOptions requestOptions = new RequestOptions();
        requestOptions.centerCrop();
        requestManager.load(data.get(position)).apply(requestOptions).into(holder.coverIv);
    }

    @Override
    public int getItemCount() {
        return data == null ? 0 : data.size();
    }

    static class HorizontalRecyclerItemViewHolder extends RecyclerView.ViewHolder{

        private ImageView coverIv;

        public HorizontalRecyclerItemViewHolder(@NonNull View itemView) {
            super(itemView);
            coverIv = itemView.findViewById(R.id.iv_cover);
        }
    }
}

File: app/src/main/java/com/horizontal/view/adapter/HorizontalRecyclerAdapter.kt
package com.horizontal.view.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.RequestManager
import com.bumptech.glide.request.RequestOptions
import com.horizontal.view.R

class HorizontalRecyclerAdapter constructor(context: Context,private val data: ArrayList<String>?,private val requestManager: RequestManager) :
        RecyclerView.Adapter<HorizontalRecyclerItemViewHolder>() {

    var inflater: LayoutInflater = LayoutInflater.from(context)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HorizontalRecyclerItemViewHolder {
        return HorizontalRecyclerItemViewHolder(inflater.inflate(R.layout.item_recycler_view, parent, false))
    }

    override fun getItemCount(): Int {
        return data?.size ?: 0
    }

    override fun onBindViewHolder(holder: HorizontalRecyclerItemViewHolder, position: Int) {
        val url = data?.get(position) ?:""

        val requestOptions = RequestOptions()
        requestOptions.centerCrop()
        requestManager.load(url).apply(requestOptions).into(holder.cover)
    }
}